internal/server/logbuffer/logbuffer.go

```diff
@@ -71,6 +73,18 @@ func New() *Buffer {
 //
 // This is safe for concurrent access.
 func (b *Buffer) Write(entries ...Entry) {
+	// We could ask Entry if it supports time, but using ingest time is
+	// pretty standard as well. So we wrap each entry in a timed entry
+	// with the time it arrived, so that later LogMerge can make sense
+	// of the total order across multiple buffers.
+	timedEntries := make([]timedEntry, len(entries))
+	for i, ent := range entries {
+		timedEntries[i] = timedEntry{
+			ingestTs: time.Now(),
+			entry:    ent,
+		}
```

Comment:
Hmmm, What if we provided a constructor for timedEntry so that we didn't have to provide the current time each time

    func NewTimedEntry(entry Entry) timeEntry {
        return &timedEntry{
            ingestTs: time.Now(),
            entry: entry
       }
   }

making this callsite a little cleaner and not requiring the consumer to provide the current time each time

      timedEntries[i] = NewTimedEntry(ent)

Reply:
There isn't a reason too, it's completely internal.

Reply:
got it, make sense!